packages/terra-infinite-list/src/InfiniteList.jsx

```diff
@@ -0,0 +1,435 @@
+    }
+
+    const contentData = InfiniteUtils.getContentData(this.contentNode);
+    const hiddenItems = InfiniteUtils.getHiddenItems(this.scrollGroups, contentData, this.boundary.topBoundryIndex, this.boundary.bottomBoundryIndex);
+    this.scrollHeight = contentData.scrollHeight;
+    this.clientHeight = contentData.clientHeight;
+
+    if (ensureUpdate || hiddenItems.topHiddenItem.index !== this.boundary.topBoundryIndex || hiddenItems.bottomHiddenItem.index !== this.boundary.bottomBoundryIndex) {
+      this.preventUpdate = true;
+      this.boundary = {
+        topBoundryIndex: hiddenItems.topHiddenItem.index,
+        hiddenTopHeight: hiddenItems.topHiddenItem.height,
+        bottomBoundryIndex: hiddenItems.bottomHiddenItem.index,
+        hiddenBottomHeight: hiddenItems.bottomHiddenItem.height,
+      };
+      this.forceUpdate();
+    }
+
+    if (!preventRequest && InfiniteUtils.shouldTriggerItemRequest(contentData)) {
+      this.triggerItemRequest();
+    }
+  }
+
+  updateScrollGroups() {
+    if (!this.contentNode) {
+      return;
+    }
+
+    let groupHeight = 0;
+    let groupIndex = 0;
+    let captureOffsetTop = true;
+    const maxGroupHeight = 1 * this.contentNode.clientHeight;
+    this.scrollGroups = [];
+    for (let i = 0; i < this.itemsByIndex.length; i += 1) {
+      const item = this.itemsByIndex[i];
+      if (this.scrollGroups[groupIndex] && item.height >= maxGroupHeight) {
+        groupHeight = 0;
+        groupIndex += 1;
+        captureOffsetTop = true;
+      }
+
+      groupHeight += item.height;
+      this.scrollGroups[groupIndex] = this.scrollGroups[groupIndex] || { items: [] };
+      this.scrollGroups[groupIndex].items.push(i);
+      this.scrollGroups[groupIndex].height = groupHeight;
+      this.itemsByIndex[i].groupIndex = groupIndex;
+      if (captureOffsetTop) {
+        this.scrollGroups[groupIndex].offsetTop = this.itemsByIndex[i].offsetTop;
+        captureOffsetTop = false;
+      }
+
+      if (groupHeight >= maxGroupHeight) {
+        groupHeight = 0;
+        groupIndex += 1;
+        captureOffsetTop = true;
+      }
+    }
+  }
+
+  updateHeight(node, index) {
+    if (node) {
+      this.itemsByIndex[index] = this.itemsByIndex[index] || {};
+      let updatedHeight = false;
+      const newHeight = node.getBoundingClientRect().height;
+      if (!this.itemsByIndex[index].height || Math.abs(this.itemsByIndex[index].height - newHeight) > 1) {
+        this.itemsByIndex[index].height = newHeight;
+        updatedHeight = true;
+      }
+      if (!this.itemsByIndex[index].offsetTop || Math.abs(this.itemsByIndex[index].offsetTop - node.offsetTop) > 1) {
+        this.itemsByIndex[index].offsetTop = node.offsetTop;
+      }
+      if (this.itemsByIndex.length === this.childCount) {
+        if (!this.scrollGroups.length) {
+          this.updateScrollGroups();
+        } else if (updatedHeight) {
+          this.adjustHeight();
+        }
+      }
+    }
+  }
+
+  adjustHeight() {
```

Comment:
This is pretty heavy. Is there a way to add a test to adjust for this? I notice there is never a call to load more items ever? If the height is lengthened would this be expected?